hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/utils/db/TestRDBStore.java

```diff
@@ -348,5 +348,40 @@ public void testGetDBUpdatesSince() throws Exception {
     }
   }
 
+  @Test
+  public void testDowngrade() throws Exception {
+
+    // Write data to current DB which has 6 column families at the time of
+    // writing this test.
+    for (String family : families) {
+      try (Table table = rdbStore.getTable(family)) {
+        byte[] key = family.getBytes(StandardCharsets.UTF_8);
+        byte[] value =
+            RandomStringUtils.random(10).getBytes(StandardCharsets.UTF_8);
+        table.put(key, value);
+      }
+    }
+    // Close current DB.
+    rdbStore.close();
+
+    // Reopen DB with the last column family removed.
+    options = new DBOptions();
+    options.setCreateIfMissing(true);
+    options.setCreateMissingColumnFamilies(true);
+    configSet = new HashSet<>();
+    List<String> familiesMinusOne = families.subList(0, families.size() - 1);
+    for(String name : familiesMinusOne) {
+      TableConfig newConfig = new TableConfig(name, new ColumnFamilyOptions());
+      configSet.add(newConfig);
+    }
+    rdbStore = new RDBStore(rdbStore.getDbLocation(), options, configSet);
+    for (String family : familiesMinusOne) {
```

Comment:
Question: If column family is missing also, but DBStore will open that column family right currently? So, do we need to check the columnFamily is opened or not also?

Reply:
Added check for the extra column family.